### PR TITLE
Nodejs/week1

### PR DESCRIPTION
**Pull request by qamarfj**

### CODE REVIEW

File: nodejs/week1/src/backend/main.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// import data here
+const meals = require("./data/meals");
+const reservations = require("./data/reservations.json");
+const reviewsjson = require("./data/reviews.json");
+
+module.exports = {
+  // return all the meals (including it's reviews)
+  getAllMealsIncludedReiews: () => {
+    //create a deep copy of Meals before including reviews in it.
+    const allMeals = JSON.parse(JSON.stringify(meals));
```
Comment (aina21) [inline nodejs/week1/src/backend/main.js:10]: why do you need to parse meals?

File: nodejs/week1/src/backend/main.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    return allMealsIncludedReiews;
+  },
+  /**return all the meals (including it's reviews)
+   *  that are cheap (you define what a cheap meal is) */
+  getAllCheapMealsIncludedReiews: () => {
+    //create a deep copy of Meals before including reviews in it.
+    const allMeals = JSON.parse(JSON.stringify(meals));
+    const allCheapMealsIncludedReiews = allMeals
+      .filter((currentMeal) => currentMeal.price < 69)
+      .map((meal) => {
```
Comment (aina21) [inline nodejs/week1/src/backend/main.js:29]: Maybe it's better to create a global variable for meals with review, and you can avoid repeating it on functions.

**Review by aina21 — APPROVED** [review]
nice job :+1:

File: nodejs/week1/src/backend/main.js (hunk 1 above)
Comment (qamarfj) [reply nodejs/week1/src/backend/main.js:10]: Thank you for your precious time for the review and valuable comments. Here I want to create a deep copy of Meals.

File: nodejs/week1/src/backend/main.js (hunk 2 above)
Comment (qamarfj) [reply nodejs/week1/src/backend/main.js:29]: ok, thanks

**Pull request merged**